velox/functions/prestosql/JsonFunctions.h

```diff
@@ -46,4 +46,20 @@ struct JsonExtractScalarFunction {
   }
 };
 
+template <typename T>
+struct JsonArrayLengthFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      int64_t& result,
+      const arg_type<Varchar>& json) {
+    auto parsedJson = folly::parseJson(json);
+    if(!parsedJson.isArray()){
+      result = int64_t(NULL);
```

Comment:
Hi Pramod, `void call(...)` indicates this function never returns null. To make it return null, write it as `bool call(...)` instead. Then return true if the current result is not null, or return false if the current result is null (and no need to assign anything to `result` if it is supposed to be null). (An explanation can be found right above the second code example here: https://facebookincubator.github.io/velox/develop/scalar-functions.html#simple-functions.)

Reply:
Thanks for sharing, modified accordingly